File: test/examples/bad-example.rb
$global_var = 5 # global_vars
SECRET_TOKEN = "xyz789" # exposed_secrets

def factorial(n) # recursion
  factorial(n - 1)
end

arr = Array.new(100) # dynamic_memory

loop do # unbounded_loops
  if arr[0]
    if arr[1] # nested_conditionals
      retry if gets.chomp.to_i > 0 # complex_flow (retry), unsafe_input
    end
  end
  sleep 1 # set_timeout
  return 1 # complex_flow (multiple returns)
rescue
  puts "Error" # try_catch, insufficient_logging
  return 2
end

File: test/examples/bad-example.php
<?php
$secretKey = "xyz123"; // exposed_secrets

function factorial($n) { // recursion
    return factorial($n - 1);
}

$arr = array(100); // dynamic_memory

while (true) { // unbounded_loops
    if ($arr[0]) {
        if ($arr[1]) { // nested_conditionals
            goto end; // complex_flow (goto)
        }
    }
    echo $_GET['input']; // unsafe_input
    sleep(1); // set_timeout
    return 1; // complex_flow (multiple returns)
    end:
    try {
        throw new Exception("Error"); // try_catch
        return 2;
    } catch (Exception $e) {
        echo "Error"; // insufficient_logging
    }
}